intel_ish: footprint: build failed with "multiple definition of `soc_early_init_hook'"

**Describe the bug**
build error for `tests/benchmarks/footprints -DCONF_FILE=prj_pm.conf`

bisected to c344771d8bf
#78235 

**To Reproduce**
```
west build -p -b intel_ish_5_4_1 tests/benchmarks/footprints -DCONF_FILE=prj_pm.conf
```

**Expected behavior**
no build error

**Logs and console output**
```
[120/129] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /__w/os.rtos.zephyr.zephyr/os.rtos.zephyr.zephyr/twister-out/intel_ish_5_4_1/tests/benchmarks/footprints/benchmark.kernel.footprints.pm/zephyr/zephyr_pre0.map 
: && /opt/toolchains/zephyr-sdk-0.16.8/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-gcc -m32  -Os -DNDEBUG -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/x86/core/offsets/offsets.c.obj  -fuse-ld=bfd  -T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/__w/os.rtos.zephyr.zephyr/os.rtos.zephyr.zephyr/twister-out/intel_ish_5_4_1/tests/benchmarks/footprints/benchmark.kernel.footprints.pm/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/arch/x86/core/libarch__x86__core.a  zephyr/lib/libc/picolibc/liblib__libc__picolibc.a  zephyr/lib/libc/common/liblib__libc__common.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  modules/hal_intel/libmodules_intel_hal.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/opt/toolchains/zephyr-sdk-0.16.8/x86_64-zephyr-elf/bin/../lib/gcc/x86_64-zephyr-elf/12.2.0/32/soft-float"  -L/__w/os.rtos.zephyr.zephyr/os.rtos.zephyr.zephyr/twister-out/intel_ish_5_4_1/tests/benchmarks/footprints/benchmark.kernel.footprints.pm/zephyr  -lgcc  -Wl,--fatal-warnings  -m32  -msoft-float  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,-no-pie  -DPICOLIBC_LONG_LONG_PRINTF_SCANF  --specs=picolibc.specs  -lc  -lgcc && cd /__w/os.rtos.zephyr.zephyr/os.rtos.zephyr.zephyr/twister-out/intel_ish_5_4_1/tests/benchmarks/footprints/benchmark.kernel.footprints.pm/zephyr && /usr/bin/cmake -E true
/opt/toolchains/zephyr-sdk-0.16.8/x86_64-zephyr-elf/bin/../lib/gcc/x86_64-zephyr-elf/12.2.0/../../../../x86_64-zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(power.c.obj): in function `soc_early_init_hook':
/__w/os.rtos.zephyr.zephyr/os.rtos.zephyr.zephyr/soc/intel/intel_ish/intel_ish5/pm/power.c:77: multiple definition of `soc_early_init_hook'; zephyr/libzephyr.a(soc.c.obj):/__w/os.rtos.zephyr.zephyr/os.rtos.zephyr.zephyr/soc/intel/intel_ish/intel_ish5/soc.c:15: first defined here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain Zephyr SDK 0.16.8
 - Commit SHA: b73c5578e3563f370d4fab0f7161a017020c114c

--- a/soc/intel/intel_ish/intel_ish5/pm/power.c
+++ b/soc/intel/intel_ish/intel_ish5/pm/power.c
@@ -69,10 +69,3 @@ void sys_arch_reboot(int type)
 	sedi_pm_reset();
 }
 #endif
-
-extern void sedi_pm_init(void);
-
-void soc_early_init_hook(void)
-{
-	sedi_pm_init();
-}

--- a/soc/intel/intel_ish/intel_ish5/soc.c
+++ b/soc/intel/intel_ish/intel_ish5/soc.c
@@ -11,9 +11,14 @@
 #include "sedi_driver_hpet.h"
 #endif
 
+extern void sedi_pm_init(void);
+
 void soc_early_init_hook(void)
 {
 #if defined(CONFIG_HPET_TIMER)
 	sedi_hpet_set_min_delay(HPET_CMP_MIN_DELAY);
+#endif
+#if defined(CONFIG_PM)
+	sedi_pm_init();
 #endif
 }

--- a/soc/st/stm32/stm32g4x/soc.c
+++ b/soc/st/stm32/stm32g4x/soc.c
@@ -20,6 +20,8 @@
 #include <stm32_ll_pwr.h>
 #endif /* PWR_CR3_UCPD_DBDIS */
 
+extern void stm32_power_init(void);
+
 /**
  * @brief Perform basic hardware initialization at boot.
  *

--- a/soc/st/stm32/stm32l0x/power.c
+++ b/soc/st/stm32/stm32l0x/power.c
@@ -72,11 +72,3 @@ void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 	 */
 	irq_unlock(0);
 }
-
-/* Initialize STM32 Power */
-void soc_early_init_hook(void)
-{
-
-	/* Enable Power clock */
-	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_PWR);
-}

--- a/soc/st/stm32/stm32l0x/soc.c
+++ b/soc/st/stm32/stm32l0x/soc.c
@@ -40,4 +40,8 @@ void soc_early_init_hook(void)
 	 * https://github.com/zephyrproject-rtos/zephyr/issues/#34324 )
 	 */
 	LL_AHB1_GRP1_EnableClock(LL_AHB1_GRP1_PERIPH_DMA1);
+#ifdef CONFIG_PM
+	/* Enable Power clock */
+	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_PWR);
+#endif
 }

